docs/manual/resource-leak-checker.tex

```diff
@@ -0,0 +1,333 @@
+  ignoring them in favor of the default ownership assumptions.
+  See Section~\ref{resource-leak-ownership}.
+\item The \<-AnoResourceAliasing> option disables resource alias support,
+  treating \<@MustCallAlias> identically to \<@PolyMustCall>.
+  See Section~\ref{resource-leak-resource-alias}.
+\item The \<-AnoCreatesMustCallFor> option disables obligation creation,
+  instead issuing errors whenever a new obligation might be created.
+  See Section~\ref{resource-leak-creates-obligation}.
+\item The \<-AcountMustCall> option prints an estimate of the number of
+  must-call obligations in the analyzed program.
+\end{itemize}
+
+
+\sectionAndLabel{Resource Leak Checker annotations}{resource-leak-annotations}
+
+The Resource Leak Checker relies on the type qualifiers of its constituent checkers:
+the Must Call Checker (Section~\ref{must-call-annotations}) and
+the Called Methods Checker (Section~\ref{called-methods-spec}). You might need
+to write qualifiers from either type hierarchy. The most common annotations from
+these checkers that you might need to write are:
+
+\begin{description}
+
+\item[\refqualclasswithparams{checker/mustcall/qual}{MustCall}{String[] value}]
+for example on an element with static type \<Object> that might contain a \<Socket>.
+
+\item[\refqualclasswithparams{checker/mustcall/qual}{InheritableMustCall}{String[] value}]
+on any classes defined in your program that have must-call obligations. See Section~\ref{must-call-on-class}.
+
+\item[\refqualclass{checker/calledmethods/qual}{EnsuresCalledMethods}] on a method in
+your code that fulfills a must-call obligation of one of its parameters or of a field.
+See Section~\ref{called-methods-ensurescalledmethods}.
+
+\end{description}
+
+The Resource Leak Checker supports annotations that express common
+aliasing patterns related to resource leaks:
+
+\begin{description}
+
+\item[\refqualclass{checker/mustcall/qual}{Owning}]
+\item[\refqualclass{checker/mustcall/qual}{NotOwning}]
+  expresses ownership.  When two aliases exist to the same Java object,
+  \<@Owning> and \<@NotOwning> indicate which of the two is responsible for
+  fulfilling must-call obligations.  For details, see
+  Section~\ref{resource-leak-ownership}.
+
+\item[\refqualclass{checker/mustcall/qual}{MustCallAlias}]
+  represents a ``resource-aliasing'' relationship.  Resource aliases are
+  distinct Java objects that nevertheless control the same resource(s):
+  fulfilling the must-call obligations of one is equivalent to
+  fulfills the obligations of the other.  For details,
+  see Section~\ref{resource-leak-resource-alias}.
+
+\end{description}
+
+The Resource Leak Checker also supports an annotations to permit re-assigning
+fields or re-opening resources:
+
+\begin{description}
+
+\item[\refqualclasswithparams{checker/mustcall/qual}{CreatesMustCallFor}{String value}]
+  is a declaration annotation that indicates that the annotated method ``resets'' the must-call
```

Comment:
Is "reset" still the right terminology to use here?  (I'm OK with it but wanted to double-check.)

Reply:
I still think it's the most intuitive way to explain what the annotation means that uses different words than the literal annotation name, which is why it's used here.

Reply:
I'm open to rephrasing this if you think it's problematic, but I'm inclined to keep it otherwise.

Reply:
It's not clear to what the obligation reset?

What about:
```
is a declaration annotation that indicates that after a call to a method with this annotation none of the must-call obligations of the expression listed in value have been met.   In other words, the annotated method ``resets'' the must-call obligations of the expression.
```